app/javascript/lib/scp-api.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @fileoverview JavaScript client for Single Cell Portal REST API
+ * 
+ * Succinct, well-documented SCP API wrappers, also enabling easy mocks
+ * 
+ * API docs: https://singlecell.broadinstitute.org/single_cell/api
+ */
+import camelcaseKeys from 'camelcase-keys';
+
+const defaultBasePath = '/single_cell/api/v1';
+
+// API endpoints that use HTTP methods other than the SCP API default
```

Comment:
I like what you've done here to let us have an API client without having to write a JS function for every single API call.  It would be nice if this could be refactored so that there was only a single place to edit the params/headers for a given function.  maybe a const overrideAttributes = { <path>: {method: 'POST', headers: {...} }

Reply:
Good points, I've incorporated them [here](https://github.com/broadinstitute/single_cell_portal_core/pull/392/files#diff-03f3b349061e103c35d7c25dc0fa9c76R1).

One of my design intents in this module is to provide one JS function per API endpoint.  The main benefit of such "endpoint functions" is convenient documentation, e.g. you can hover over the use of `fetchAuthCode` in `DownloadButton.js` and get its full doc string, including examples.

In our chat, you noted how this module mixed a configuration-based approach (`otherMethods`, `otherHeaders`) with a function-based approach (having paths in endpoint-specific functions).  You suggested picking only one.  So I consolidated to the function-based approach.

To enable easier extension, I applied your `const overrideAttributes` suggestion as a new `init` parameter.  This matches [`init` from `fetch`](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Syntax), and can override the new `defaultInit` when the endpoint function needs something different from the SCP API boilerplate.  [`fetchAuthCode`](https://github.com/broadinstitute/single_cell_portal_core/pull/392/files#diff-03f3b349061e103c35d7c25dc0fa9c76R36) gives an example.